### ISSUE TITLE
Remove package dependency from Kestrel to Kestrel.Transport.Libuv

### ISSUE THREAD

**Issue by natemcmaster** [aspnet/KestrelHttpServer#2361]
Related issue: #2360 

Remove the package dependency from Microsoft.AspNetCore.Server.Kestrel to Microsoft.AspNetCore.Server.Kestrel.Transport.Libuv.

This will break those who only depend on the Microsoft.AspNetCore.Server.Kestrel package _and_ who call `WebHostBuilderLibuvExtensions.UseLibuv()`. These users can fix their projects by re-adding the dependency to libuv.
```xml
<PackageReference Include="Microsoft.AspNetCore.Server.Kestrel.Transport.Libuv" Version="2.1.0" />
```

To minimize the number of 2.0 users that this will break, we should keep Microsoft.AspNetCore.Server.Kestrel.Transport.Libuv in the dependency graph of Microsoft.AspNetCore.All.

[Design meeting notes from March 2, 2018](https://gist.github.com/natemcmaster/43a3980b724fb0b167844fb574df75dc)

/cc @halter73 @davidfowl @muratg

**Comment by muratg** [comment]
@natemcmaster Libuv package is directly in All I assume?  So if you're using the metapackage, you wouldn't be broken?  (As long as we don't remove it)

**Comment by natemcmaster** [comment]
Correct you won't be broken. No, .All doesn't directly reference "Libuv", but it does directly reference "Microsoft.AspNetCore.Server.Kestrel.Transport.Libuv", so you still get the Libuv package transitively in .All

**Comment by halter73** [comment]
Fixed by #2392 and aspnet/Universe#959
